snuba/migrations/snuba_migrations/events/0010_groupedmessages_onpremise_compatibility.py

```diff
@@ -0,0 +1,89 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, UInt
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+
+
+TABLE_NAME = "groupedmessage_local"
+TABLE_NAME_NEW = "groupedmessage_local_new"
+TABLE_NAME_OLD = "groupedmessage_local_old"
+
+
+def fix_order_by() -> None:
+    cluster = get_cluster(StorageSetKey.EVENTS)
+    clickhouse = cluster.get_query_connection(ClickhouseClientSettings.MIGRATE)
+    database = cluster.get_database()
+
+    new_primary_key = "project_id, id"
+    old_primary_key = "id"
+
+    ((curr_primary_key,),) = clickhouse.execute(
+        f"SELECT primary_key FROM system.tables WHERE name = '{TABLE_NAME}' AND database = '{database}'"
+    )
+
+    if curr_primary_key != old_primary_key:
+        return
+
+    # There shouldn't be any data in the table yet
+    assert (
+        clickhouse.execute(f"SELECT COUNT() FROM {TABLE_NAME} FINAL;")[0][0] == 0
+    ), f"{TABLE_NAME} is not empty"
+
+    new_order_by = f"ORDER BY ({new_primary_key})"
+    old_order_by = f"ORDER BY {old_primary_key}"
+
+    ((curr_create_table_statement,),) = clickhouse.execute(
+        f"SHOW CREATE TABLE {database}.{TABLE_NAME}"
+    )
+
+    new_create_table_statement = curr_create_table_statement.replace(
+        TABLE_NAME, TABLE_NAME_NEW
+    ).replace(old_order_by, new_order_by)
```

Comment:
I could duplicate the create table operation in migration 0007 in here if you think that is better, I thought it'd be easier to avoid copying things over and I'm fairly confident there haven't been any other changes to this table except this one.

Reply:
mh ok, I think we should make it simpler for the future to build the table from the official schema and not from the existing one. This pattern of taking the SHOW CREATE TABLE and tweaking the string is quite weak. A lot of things can go wrong. And we would have a much more solid guarantee that the schema is consistent with what we expect.